public/editor-client/src/api/index.ts

```diff
@@ -1206,3 +1217,175 @@ export const updateGlobalBlocks = async (
 };
 
 //#endregion
+
+//#region Default Templates
+
+export const getDefaultKits = async (
+  url: string,
+  id: string
+): Promise<{
+  blocks: Kit[];
+  categories: { slug: string; title: string }[];
+  styles: Style;
+}> => {
+  const fullUrl = makeUrl(url, {
+    project_id: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    return {
+      blocks: res.collections,
+      categories: res.categories,
+      styles: res.styles
+    };
+  }
+
+  throw new Error(t("Failed to load kits"));
+};
+
+export const getKitData = async (
+  url: string,
+  kitId: string,
+  id: string
+): Promise<DefaultBlock> => {
+  const fullUrl = makeUrl(url, {
+    project_id: kitId,
+    page_slug: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    const parsedResult = mPipe(
+      Obj.readKey("collection"),
+      Arr.read,
+      (res) => res[0] as { pageData: string },
+      Obj.readKey("pageData"),
+      (r) => Json.read(r) as { items: string },
+      Obj.readKey("items"),
+      Arr.read,
+      (res) => res[0] as DefaultBlock
+    )(res);
```

Comment:
create an interface for your `res` constant to avoid casting:
```ts
interface ResponseType {
  collection: { pageData: string }[] /* and next for other nested keys */
}

const res: ResponseType = await response.json();
```